File: leetcode-submissions/1208_Get_Equal_Substrings_Within_Budget.ts
function equalSubstring(s: string, t: string, maxCost: number): number {
    let i : number=0 , j:number =0 , currCost:number =0, maxWindow:number = Number.MIN_SAFE_INTEGER ;
    const n :number = s.length;

    while(j<n){
        currCost +=Math.abs(s[j].charCodeAt(0)-t[j].charCodeAt(0));
        while(currCost>maxCost){
            currCost -=Math.abs(s[i].charCodeAt(0)-t[i].charCodeAt(0));
            i++;
        }
    maxWindow = Math.max(maxWindow, j-i+1);
        j++;
    }
    return maxWindow;
};

File: leetcode-submissions/1208_Get_Equal_Substrings_Within_Budget.cpp
class Solution {
public:
    int equalSubstring(string s, string t, int maxCost) {
        
        // a b c d 
        // b c d f
        // 1 1 1 2
        // 1 2 3 
        // l=0 2-0+1 =3
        int currcost=0;
        int left=0;
        int result=0;

        for(int i=0;i<s.size();i++){
            currcost+=abs(s[i]-t[i]);
            while(currcost>maxCost){
                currcost-=abs(s[i]-t[i]);
                left++;
            }
            result=max(result,i-left+1);

        }
return result;
    }
};